### PR TITLE
User can view the details of a products

### PR DESCRIPTION
**Pull request by malmossa**
image

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -34,6 +34,31 @@ app.get('/api/products', (req, res, next) => {
...
+    select *
+      from "products"
+      where "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next();
```
Comment (scbowler) [inline server/index.js:54]: Here you want to call `next` with a new client error:

```JAVASCRIPT
const productError = new ClientError('Unable to find product with supplied ID', 400);
return next(productError);
```

You can put whatever you want for the actual error message

**Review by scbowler — CHANGES_REQUESTED** [review]
Everything looks good, just need one fix with your error handling, see below comment.

File: server/index.js (hunk 1 above)
Comment (malmossa) [reply server/index.js]: image

**Review by scbowler — APPROVED** [review]
Looks good 👍

**Pull request merged**